src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/CorrelationIdLookupHelper.cs

```diff
@@ -0,0 +1,199 @@
+namespace Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Globalization;
+    using System.Threading.Tasks;
+    
+    using Microsoft.ApplicationInsights.AspNetCore.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+#if NET451
+    using System.IO;
+    using System.Net;
+#else
```

Comment:
The codebase, in fact, supports both net451 & netstandard1.6 as in project.json:
-
    "frameworks": {
        "net451": {
            "dependencies": {
                "Microsoft.ApplicationInsights.DependencyCollector": "2.3.0-beta3",
                "Microsoft.ApplicationInsights.PerfCounterCollector": "2.3.0-beta3",
                "Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel": "2.3.0-beta3"
            }
        },
        "netstandard1.6": {
            "dependencies": {
                "NETStandard.Library": "1.6.0",
                "System.Net.NameResolution": "4.0.0"
            }
        }
    }
-
Since this is a general helper method, I think we just keep it working for both target frameworks.